tests/by-util/test_mv.rs

```diff
@@ -1716,4 +1752,215 @@ mod inter_partition_copying {
+        let scene = TestScenario::new(util_name!());
+        let at = &scene.fixtures;
+        at.write("a", "file_contents");
+
+        // create a folder in another partition.
+        let other_fs_tempdir =
+            TempDir::new_in("/dev/shm/").expect("Unable to create temp directory");
+
+        // mv to other partition
+        scene
+            .ucmd()
+            .arg("-v")
+            .arg("a")
+            .arg(other_fs_tempdir.path().to_str().unwrap())
+            .succeeds()
+            .stdout_contains(format!(
+                "copied 'a' -> {}",
+                other_fs_tempdir.path().join("a").to_string_lossy().quote()
+            ))
+            .stdout_contains("removed 'a'");
+
+        at.write("a", "file_contents");
+
+        // mv to other partition
+        scene
+            .ucmd()
+            .arg("-vb")
+            .arg("a")
+            .arg(other_fs_tempdir.path().to_str().unwrap())
+            .succeeds()
+            .stdout_contains(format!(
+                "copied 'a' -> {} (backup: {})",
+                other_fs_tempdir.path().join("a").to_string_lossy().quote(),
+                other_fs_tempdir.path().join("a~").to_string_lossy().quote()
+            ))
+            .stdout_contains("removed 'a'");
+    }
+
+    #[test]
+    fn test_inter_partition_copying_folder_without_read_permission_to_subfolders() {
+        let scene = TestScenario::new(util_name!());
+        let at = &scene.fixtures;
+        at.mkdir_all("a/b/c");
+        at.write("a/b/d", "d");
+        at.write("a/b/c/e", "e");
+        at.mkdir_all("a/b/f");
+        at.write("a/b/f/g", "g");
+        at.write("a/b/h", "h");
+        at.set_mode("a/b/f", 0);
+        // create a folder in another partition.
+        let other_fs_tempdir =
+            TempDir::new_in("/dev/shm/").expect("Unable to create temp directory");
+
+        // mv to other partition
+        scene
+            .ucmd()
+            .arg("-v")
+            .arg("a")
+            .arg(other_fs_tempdir.path().to_str().unwrap())
+            .fails()
+            // check erorr occured
+            .stderr_contains(format!("cannot access 'a/b/f': Permission denied"));
+
+        // make sure mv kept on going after error
+        assert_eq!(
+            read_to_string(other_fs_tempdir.path().join("a/b/h"),)
+                .expect("Unable to read other_fs_file"),
+            "h"
+        );
+
+        // make sure mv didn't remove src because an error occured
+        at.dir_exists("a");
+        at.file_exists("a/b/h");
+    }
+
+    #[test]
+    fn test_inter_partition_copying_folder_without_write_permission_to_subfolders() {
+        let scene = TestScenario::new(util_name!());
+        let at = &scene.fixtures;
+        at.mkdir_all("a/b/c");
+        at.write("a/b/d", "d");
+        at.write("a/b/c/e", "e");
+        at.mkdir_all("a/b/f");
+        at.write("a/b/f/g", "g");
+        at.write("a/b/h", "h");
+        at.set_mode("a/b/f",0o555 );
```

Comment:
Formatting error. Why wasn't this caught by CI fmt?
```suggestion
        at.set_mode("a/b/f", 0o555);
```